### PR TITLE
Configure the Android NDK

### PR DESCRIPTION
**Pull request by asajeffrey**
This patch adds configuration for the Android NDK directory layout. It's used when building for the Magic Leap, which uses a very similar, but not quite identical, NDK.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/mozjs/158)

### CODE REVIEW

**Comment by jdm** [comment]
Please submit these upstream, as well.

File: etc/patches/configure-override-android-ndk-paths.patch [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
++
+     if 'mips' in target.cpu:
+         target_dir_name = 'mips'
+     elif 'aarch64' == target.cpu:
+@@ -257,7 +273,7 @@ option(env='STLPORT_CPPFLAGS',
+ @imports(_from='os.path', _import='isdir')
+ def stlport_cppflags(value, ndk, _):
+     if value and len(value):
+-        return value.split()
++        return value[0].split()
```
[the remark this answers is not in the document]
Comment (asajeffrey) [reply etc/patches/configure-override-android-ndk-paths.patch]: Good question. I made the edit to bring `STLPORT_CPPFLAGS` in line with the other configurations, which use `value[0]`. Building again without this change still works, so I'll undo it.

[the remark this answers is not in the document]
Comment (asajeffrey) [reply etc/patches/configure-override-android-ndk-paths.patch]: Ah, try again, I was building with the wrong version. Without this change, we get:
```
  File "/Users/ajeffrey/.cargo/git/checkouts/mozjs-ccfd509d897f834d/f3d2e03/mozjs/build/moz.configure/android-ndk.configure", line 276, in stlport_cppflags
    return value.split()
AttributeError: 'PositiveOptionValue' object has no attribute 'split'
make: *** [maybe-configure] Error 1
```

**Comment by asajeffrey** [comment]
I'll upstream these changes. @bors-servo r=jdm

**Comment by jdm** [comment]
@bors-servo r-
Homu doesn't think that travis exists in this repo for some reason.

**Pull request merged**